hackathon/views.py

```diff
@@ -378,6 +368,17 @@ def view_hackathon(request, hackathon_id):
     return render(request, "hackathon/hackathon_view.html", context)
 
 
+def view_hackathon_public(request, hackathon_id):
+    """ A limited view of the hackathon page for the public """
+    hackathon = get_object_or_404(Hackathon, pk=hackathon_id)
+    if hackathon.status == 'deleted':
+        messages.error(request, 'This hackathon does not exist.')
+        return redirect(reverse('home'))
```

Comment:
I generally try to avoid using the `HTTP_REFERER` for the redirect because it can be empty. I have updated this to use it with a fallback to home though.

Reply:
I generally try to avoid using the `HTTP_REFERER` for the redirect because it can be empty. I have updated this to use it with a fallback to home though.

Reply:
Oh nice! Nice to include the fallback option, too!